src/components/HistoryTables/MessageHistory/Detail/index.tsx

```diff
@@ -50,26 +42,33 @@ export default function MessageDetail(props: MessageDetailProps) {
   const { isProd } = useEnvironment()
   const logger = useLogger()
 
-  const { data: gasQuery, error: gasStateError } = useGasCostQuery({
-    variables: { cid },
-    pollInterval: 10000
-  })
-  const { data: msgRcptQuery, error: msgRcptError } = useMessageReceiptQuery({
-    variables: { cid },
-    pollInterval: 10000
-  })
+  const { data: stateReplayQuery, error: stateReplayError } =
+    useStateReplayQuery({
+      variables: { cid: props.cid },
+      pollInterval: 30000,
+      fetchPolicy: 'cache-first',
+      // give the message time to execute on-chain before fetching
+      skip: !props.cid || props.confirmations < 2
+    })
 
   const transactionFee = useMemo<string>(() => {
     if (pending) return 'Pending...'
-    const cost = gasQuery?.gascost?.totalCost
+    const cost = stateReplayQuery?.stateReplay?.gasCost?.totalCost
     return cost
       ? `${makeFriendlyBalance(new FilecoinNumber(cost, 'attofil'))} FIL`
       : 'Calculating...'
-  }, [gasQuery?.gascost?.totalCost, pending])
+  }, [stateReplayQuery?.stateReplay?.gasCost?.totalCost, pending])
 
   const gasUsed = useMemo(
-    () => gasQuery?.gascost?.gasUsed || 0,
-    [gasQuery?.gascost]
+    () => stateReplayQuery?.stateReplay?.gasCost?.gasUsed || 0,
+    [stateReplayQuery?.stateReplay?.gasCost]
+  )
+
+  const executionTrace = useMemo(
+    () =>
+      (stateReplayQuery?.stateReplay?.executionTrace
+        ?.executionTrace as unknown as ExecutionTrace) || null,
```

Comment:
Since the `executionTrace` comes back as a string and we JSON.parse it before it gets cached, ts isn't happy that i'm converting the types to the parsed ExecutionTrace type. This is ugly, but wasn't sure of the "proper" workaround here (even though i thought we would have the same problem with `message.params`